File: Two Sum/Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        book = {}
        for i in range(len(nums)):
            if target - nums[i] in book:
                return [book[target - nums[i]], i]
            else:
                book[nums[i]] = i
        

File: Two Sum/Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> book = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i += 1) {
            if (book.containsKey(target - nums[i])) {
                int[] ans = {book.get(target - nums[i]), i};
                return ans;
            } else {
                book.put(nums[i], i);
            }
        }
        int[] ans = {};
        return ans;
    }
}
